mods/ra/maps/negotiations/negotiations.lua

```diff
@@ -0,0 +1,1237 @@
+			return actor.HasProperty("StartBuildingRepairs")
+		end)
+
+		if #buildings == 0 then
+			return
+		end
+
+		Utils.Do(buildings, function(building)
+			building.Sell()
+		end)
+
+		Trigger.OnAllRemovedFromWorld(buildings, function()
+			GroupIdleHunt(player.GetGroundAttackers())
+		end)
+	end)
+end
+
+SignalAllies = function()
+	Greece.MarkCompletedObjective(SignalForReinforcements)
+	SpawnFlare(TanyaDrop.Location, DateTime.Minutes(4))
+	Trigger.AfterDelay(DateTime.Seconds(1), ReinforceAllies)
+
+	if General.IsDead then
+		OrderBadGuyHunters()
+		return
+	end
+	Trigger.OnKilled(General, OrderBadGuyHunters)
+end
+
+GroupAttackMove = function(actors, location, closeEnough)
+	Utils.Do(actors, function(actor)
+		if actor.IsDead then
+			return
+		end
+		actor.AttackMove(location, closeEnough or 0)
+	end)
+end
+
+GroupIdleHunt = function(actors)
+	Utils.Do(actors, IdleHunt)
+end
+
+-- Patrol along path. Pause at each location until others arrive.
+-- At final location, call onFinished(actors) or loop and restart.
+GroupTightPatrol = function(actors, path, looped, pauseTime, onFinished)
+	pauseTime = pauseTime or 0
+	local paused = false
+	local goal = 1
+	local arrived = { }
+	local patrollers = Utils.Where(actors, IsAlive)
+
+	for id = 1, #patrollers do
+		Trigger.OnIdle(patrollers[id], function(actor)
+			if paused then
+				return
+			end
+
+			if not arrived[id] then
+				actor.AttackMove(path[goal], 2)
+				actor.CallFunc(function()
+					arrived[id] = true
+				end)
+				return
+			end
+
+			if not AreAllIdleOrDead(patrollers) then
+				return
+			end
+
+			-- All live actors have arrived and are idle.
+			paused = true
+			Trigger.AfterDelay(pauseTime, function()
+				paused = false
+			end)
+
+			arrived = { }
+			local nextLocation, patrolFinished = NextPatrolLocation(goal, #path, looped)
+
+			if patrolFinished then
+				ClearTightPatrol(patrollers, onFinished)
+				return
+			end
+
+			goal = nextLocation
+		end)
+	end
+end
+
+NextPatrolLocation = function(current, final, looped)
+	if current < final then
+		return current + 1, false
+	end
+
+	if looped then
+		return 1, false
+	end
+
+	return current, true
+end
+
+ClearTightPatrol = function(actors, onFinished)
+	Utils.Do(actors, function(actor)
+		if actor.IsDead then
+			return
+		end
+
+		Trigger.Clear(actor, "OnIdle")
+	end)
+
+	if not onFinished then
+		return
+	end
+
+	Trigger.AfterDelay(1, function()
+		onFinished(actors)
+	end)
+end
+
+GroupHuntOnDamaged = function(actors, targetPlayer)
+	local alerted = false
+	Utils.Do(actors, function(actor)
+		Trigger.OnDamaged(actor, function(_, attacker)
+			if alerted or attacker.Type == "dtrk" or not (attacker.Owner == targetPlayer) then
+				return
+			end
+			alerted = true
+
+			Utils.Do(actors, function(hunter)
+				if hunter.IsDead or not hunter.HasProperty("Hunt") then
+					return
+				end
+				-- Patrolling groups will need to halt.
+				hunter.Stop()
+				hunter.Hunt()
```

Comment:
`IdleHunt`?

Reply:
Some of these will be groups using OnIdle to form up on a patrol location, wait until all live members of the group are there, then form up on the next location. A Trigger.ClearAll followed by IdleHunt might be worth doing, though.